UnityProject/Assets/Scripts/Core/Lighting/LightAnimator.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using Logs;
+using Mirror;
+using UnityEngine;
+
+namespace Core.Lighting
+{
+	public class LightAnimator : NetworkBehaviour, IRightClickable
+	{
+		[SerializeField] private GameObject animationsHolder;
+		[SyncVar, SerializeField] private int activeAnimationID = -1;
+		private List<ILightAnimation> animations = new List<ILightAnimation>();
+		public ILightAnimation ActiveAnimation { get; private set; } = null;
+
+		private void Awake()
+		{
+			animations.AddRange(animationsHolder.GetComponents<ILightAnimation>());
+		}
+
+		public override void OnStartClient()
+		{
+			base.OnStartClient();
+			if (activeAnimationID != -1) PlayAnim(activeAnimationID);
+		}
+
+		public void StopAnim()
+		{
+			if (ActiveAnimation == null) return;
+			ActiveAnimation.StopAnimation();
+			ActiveAnimation = null;
+			activeAnimationID = -1;
+		}
+
+		[Client]
+		public void PlayAnim(ILightAnimation anim)
+		{
+			StopAnim();
+			ActiveAnimation = anim;
+			ActiveAnimation.StartAnimation();
+		}
+
+		public void PlayAnim(int animID)
+		{
+			foreach (var anim in animations)
+			{
+				if (anim.ID == animID) PlayAnim(anim);
+				break;
+			}
+		}
+
+		[ClientRpc]
+		public void PlayAnimNetworked(int animID)
```

Comment:
This is one shot?

Reply:
wdym?

Reply:
as in  can it be a continuous animation

Reply:
That's up to the animation scripts themselves, but all animations currently are all continuous.

Reply:
should be a synchvar then

Reply:
done

Reply:
would be best to name it as play animation one shot

Reply:
no animations are one shot

Reply:
then remove the RPC

Reply:
done